sklearn/cluster/mean_shift_.py

```diff
@@ -387,7 +387,12 @@ def fit(self, X, y=None):
         X : array-like, shape=[n_samples, n_features]
             Samples to cluster.
         """
-        X = check_array(X)
+
+        try:
```

Comment:
`fit` is fine `check_array` already checks that the matrix is not sparse and gives a meaningful error message. 

Do not modify `fit` but the `estimate_bandwith` function by just adding `X = check_array(X)`. Amend the test accordingly.